### PR TITLE
[NTUSER] Add multiple hwnd/pwnd to one DC

### PR DESCRIPTION
**Pull request by turican0**
## Purpose

While searching for bugs in GetDCEx, I found that reactos now only saves the first hwnd/pwnd to DC. Based on tests with windows, I found that this is not the correct implementation. E.g. when a DC exists for two hwnd and the original hwnd disappears, in many cases the hwnd for that DC should change to the next hwnd. Before this patch the extended https://github.com/reactos/reactos/pull/7296 show 40 failures, after this patch 19/21 failures(two errors only appear sometimes).

## Proposed changes

In dce.h I changed the format of the DCE header to listPwndCurrent instead of hwndCurrent, and removed the unused items (if anyone thinks they make sense, we can return them).
I added functions to windc.c that work with the given sheet, add/remove/delete/return the given hwnd/pwnd in the DC sheet and replaced them at the given places in the code.

### CODE REVIEW

File: win32ss/user/ntuser/windc.c [hunk 1]
```diff
@@ -488,9 +680,9 @@ UserGetDCEx(PWND Wnd OPTIONAL, HANDLE ClipRegion, ULONG Flags)
          Dce = DceAllocDCE(NULL, DCE_CACHE_DC);
       }
       if (Dce == NULL) return NULL;
-
-      Dce->hwndCurrent = (Wnd ? UserHMGetHandle(Wnd) : NULL);
-      Dce->pwndOrg = Dce->pwndClip = Wnd;
+
```
Comment (JoachimHenze) [inline win32ss/user/ntuser/windc.c:680]: Strip EOL whitespace. Best is to use the NPP feature, in case this is not the only one.
```suggestion

```

**Comment by HBelusca** [comment]
> and removed the unused items (if anyone thinks they make sense, we can return them).

OK so you made changes without knowing why these were here, so please revert.
In particular they are there so as to have a layout compatible with Windows' one: https://reactos.org/wiki/Techwiki:Win32k/DCE

**Comment by HBelusca** [comment]
> E.g. when a DC exists for two hwnd [...]

Is this something that is actually documented somewhere?

**Comment by turican0** [comment]
Thanks, you are right, the structure comes from Windows. I'll try to think how to solve it without changing it. I'll close the PR because I have to go about it in a completely different way. Other than the multiple HWND option didn't make sense to me based on testing, but I'll test on XP to see if they are identical.

**Pull request closed without merging**